addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputWithPlayControlXML.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * Example:
+ *
+ * InputWithPlayControl menu = new InputWithPlayControl("NET_RADIO", comObject);
+ * menu.goToPath(menuDir);
+ * menu.selectItem(stationName);
+ *
+ * No state will be saved in here, but in {@link PlayInfoState} and
+ * {@link PresetInfoState} instead.
+ *
+ * @author David Graeff
+ */
+public class InputWithPlayControlXML implements InputWithPlayControl {
+    protected final WeakReference<AbstractConnection> comReference;
+
+    protected final String inputID;
+
+    public static final int PRESET_CHANNELS = 40;
+
+    private PlayInfoStateListener observer;
+
+    /**
+     * Create a InputWithPlayControl object for altering menu positions and requesting current menu information as well
+     * as controlling the playback and choosing a preset item.
+     *
+     * @param inputID The input ID like USB or NET_RADIO.
+     * @param com The Yamaha communication object to send http requests.
+     */
+    public InputWithPlayControlXML(String inputID, AbstractConnection com, PlayInfoStateListener observer) {
+        this.inputID = inputID;
+        this.comReference = new WeakReference<AbstractConnection>(com);
+        this.observer = observer;
+    }
+
+    /**
+     * Wraps the XML message with the inputID tags. Example with inputID=NET_RADIO:
+     * <NETRADIO>message</NETRADIO>.
+     *
+     * @param message XML message
+     * @return
+     */
+    protected String wrInput(String message) {
+        return "<" + inputID + ">" + message + "</" + inputID + ">";
+    }
+
+    /**
+     * Updates the playback information
+     *
+     * @throws Exception
+     */
+    @Override
+    public void update() throws IOException, ReceivedMessageParseException {
+        if (observer == null) {
+            return;
+        }
+
+        AbstractConnection com = comReference.get();
+        String response = com.sendReceive(wrInput("<Play_Info>GetParam</Play_Info>"));
+        Document doc = XMLUtils.xml(response);
+        if (doc.getFirstChild() == null) {
+            throw new ReceivedMessageParseException("<Play_Info>GetParam failed: " + response);
+        }
+
+        PlayInfoState msg = new PlayInfoState();
+
+        Node playbackInfoNode = XMLUtils.getNode(doc.getFirstChild(), "Play_Info/Playback_Info");
+        msg.playbackMode = playbackInfoNode != null ? playbackInfoNode.getTextContent() : "";
+
+        Node metaInfoNode = XMLUtils.getNode(doc.getFirstChild(), "Play_Info/Meta_Info");
+        Node sub;
+
+        if (inputID.equals("TUNER")) {
+            sub = XMLUtils.getNode(metaInfoNode, "Radio_Text_A");
+            msg.station = sub != null ? sub.getTextContent() : "";
+        } else {
+            sub = XMLUtils.getNode(metaInfoNode, "Station");
+            msg.station = sub != null ? sub.getTextContent() : "";
+        }
+
+        sub = XMLUtils.getNode(metaInfoNode, "Artist");
+        msg.artist = sub != null ? sub.getTextContent() : "";
+
+        sub = XMLUtils.getNode(metaInfoNode, "Album");
```

Comment:
@martinvw  Yes, I actually refactored some of this in my other branch ([2574_yamaha_spotify](https://github.com/zarusz/openhab2-addons/tree/2574_yamaha_spotify)) that will add Spotify feature #2574. I intended to merge it to this PR. Do you suggest I merge to this PR or you could wait with this until another PR ?